lib/src/pubspec.dart

```diff
@@ -22,8 +22,19 @@ class Pubspec {
   final Version version;
 
   final String description;
+
+  /// This should be a URL pointing to the website for the package.
   final String homepage;
 
+  /// Optional field to specify the source code repository of the package.
+  /// Useful when a package has both a home page and a repository.
+  final String repository;
```

Comment:
I've changed it to `Uri`. I don't think it would cause issues, and even if it does, we already handled the generic yaml-parsing-issue cases, this would be probably in that path too.